components/match2/wikis/ageofempires/legacy/match_group_legacy_default.lua

```diff
@@ -0,0 +1,64 @@
+---
+-- @Liquipedia
+-- wiki=ageofempires
+-- page=Module:MatchGroup/Legacy/Default
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+
+local MatchGroupLegacy = Lua.import('Module:MatchGroup/Legacy')
+
+---@class AgeofEmpiresMatchGroupLegacyDefault: MatchGroupLegacy
+local MatchGroupLegacyDefault = Class.new(MatchGroupLegacy)
+
+---@return table
+function MatchGroupLegacyDefault:getMap()
+	local map = {
+		['$notEmpty$'] = 'map$1$',
+		map = 'map$1$',
+		mode = 'map$1$mode',
+		winner = 'map$1$win',
+		civs1 = 'map$1$p1civ',
+		civs2 = 'map$1$p2civ',
+		vod = 'vodgame$1$',
+		date = 'date$1$',
+	}
+
+	if self.bracketType == 'team' then
+		Table.mergeInto(map, {
+			players1 = 'map$1$t1players',
+			players2 = 'map$1$t2players',
+			civs1 = 'map$1$t1civs',
+			civs2 = 'map$1$t2civs',
+		})
+	end
+
+	return map
+end
+
+---@param details table
+---@param mapIndex number
+---@return table?
+function MatchGroupLegacyDefault:handleMap(details, mapIndex)
+	local blueprint = self:_getMap(mapIndex)
+	if Logic.isEmpty(details[blueprint['$notEmpty$']]) then
+		return nil
+	end
+
+	blueprint = Table.copy(blueprint)
+	blueprint['$notEmpty$'] = nil
+
+	return self:_copyAndReplace(blueprint, details, true)
```

Comment:
If I add processMap to MGI, player civs,names and flags don't appear

Reply:
this still an issue?

Reply:
I need to test it

Reply:
Same problem occurs, but we can keep it, because we are overriding the function to also check if the winner is set

Reply:
Unless you need to add processMap to MGI, skip it. Planning on removing it anyway.

Reply:
Ok, but this part needs to remain as it is.